packages/react/src/components/authentication/AuthFlow.tsx

```diff
@@ -88,21 +88,25 @@ export const AuthFlow: FC = () => {
       case BlockTypes.Completed:
         return null;
     }
-  }, [currentScreen]);
+  };
 
-  const render = useMemo(() => {
+  const screen = (() => {
     switch (initState) {
       case InitState.Initializing:
         return <Loading />;
       case InitState.Failed:
         return <ComponentUnavailableError />;
       case InitState.Success:
-        return screenComponent;
+        return screenComponent();
     }
-  }, [initState, screenComponent]);
+  })();
 
   // Render the component if it exists, otherwise a fallback or null
-  return (
+  return initState === InitState.Failed ? (
```

Comment:
Now we handle the InitState.Initializing and InitState.Failed states twice right? (they are handled in the switch and in the if statement => isn't the switch enough?)

Reply:
Removed the render function. We don't want to show the loader inside the container as it doesn't look good in the UI that is why we only use container for main screen components.